int32 agent_id
int32 zone_id